### PR TITLE
bugfix: do not publish last will when authentication failed

### LINKED ISSUE #8886: LWT is sent when a connection is rejected
### What happened?

I try to connect a client that doesn't have permissions, setting the last will message option, and althought the validation works perfectly, the broker publish the LWT message.
I think that it is a bug, because the client wasn't connected to the broker.

### What did you expect to happen?

I expect that the broker doesn't send LWT message because the client was not connected.

### How can we reproduce it (as minimally and precisely as possible)?

Try to connect to a broker with the MQTTX.app , set the LWT options ( topic:any that has a subscriber to could see the LWT that is published, payload:any, etc.. ) and set wrong client's options (username, clientid, password).
The subscriber will receive a LWT message of the client that is not connected.

### Anything else we need to know?

_No response_

### EMQX version

sysdescr  : EMQX
version   : 5.0.2
datetime  : 2022-08-29T11:37:37.570495507+00:00
uptime    : 2 days, 12 hours, 46 minutes, 34 seconds

### OS version

```console
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

$ uname -a
Linux 69ee644bc5ed 5.10.104-linuxkit #1 SMP Thu Mar 17 17:08:06 UTC 2022 x86_64 GNU/Linux
```

### Log files

<details>

</details>

### PR DESCRIPTION
**Pull request by thalesmg**
fixes #8886

### CODE REVIEW

File: apps/emqx/test/emqx_broker_SUITE.erl [hunk 1]
```diff
@@ -733,6 +733,62 @@ t_handle_in_empty_client_subscribe_hook(Config) when is_list(Config) ->
...
+    ]),
+    {error, _} = emqtt:connect(Publisher),
+
+    receive
+        {publish, #{
+            topic := <<"lwt">>,
+            payload := <<"should not be published">>
+        }} ->
+            ct:fail("should not publish will message")
+    after 1_000 ->
```
Comment (zmstone) [inline apps/emqx/test/emqx_broker_SUITE.erl:786]: Maybe use a tp and expect a not sent trace point?

Comment (zmstone) [reply apps/emqx/test/emqx_broker_SUITE.erl:786]: Or wait/assert connection process is gone

Comment (thalesmg) [reply apps/emqx/test/emqx_broker_SUITE.erl]: Ok, I'll wait for the publisher process to die and check we didn't receive any messages.

Comment (thalesmg) [reply apps/emqx/test/emqx_broker_SUITE.erl]: :heavy_check_mark:

**Review by ieQu1 — APPROVED** [review]

**Pull request merged**